Date/Date.cpp

```diff
@@ -0,0 +1,61 @@
+#include <iostream>
+#include "Date.h"
+
+using namespace std;
+
+void Date::validate(int day, int month, int year) {
+    if ( day < 1 || day > 31 ) {
+        throw InvalidDate("Invalid Day!");
+    }
+    if ( month < 1 || month > 12 ) {
+        throw InvalidDate("Invalid Month!");
+    }
+    if ( year < 0 ) {
+        throw InvalidDate("Invalid Year!");
+    }
+    if ( month == 2 || month == 4 || month == 6 || month == 9 || month == 11 ) {
```

Comment:
It is better to create array int daysInMonth = { 0, 31, 30, 28 ... } and then check like;

if ( day > daysInMonth[month]) { ... }